components/Button.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { type ReactNode } from "react";
+
+interface ButtonProps
+  extends React.PropsWithChildren<
+    React.ButtonHTMLAttributes<HTMLButtonElement>
+  > {
+  icon?: ReactNode;
+}
+
+const buttonStyles =
+  "flex items-center gap-2 px-3 py-2 w-max rounded transition-all hover:shadow-md";
+
+export default function Button({ icon, children, ...delegated }: ButtonProps) {
+  return (
+    <button
+      className={`${buttonStyles} bg-secondary-blue-500 text-white hover:bg-secondary-blue-700`}
+      {...delegated}
+    >
+      {icon}
+      <span>{children}</span>
+    </button>
+  );
+}
+
+export function ButtonSecondary({ icon, children, ...delegated }: ButtonProps) {
```

Comment:
This might just be a me thing, but I feel like I'd prefer an API thats a single `Button` component with a `variant` prop that's optional and is either `"secondary"` or `"tertiary"`. Do you prefer three different components?